proxy/grpcproxy/watcher_single.go

```diff
@@ -53,9 +52,8 @@ func (ws watcherSingle) run() {
 	defer close(ws.donec)
 
 	for wr := range ws.ch {
-		ws.rev = wr.Header.Revision
+		ws.lastSeenRev = wr.Header.Revision
```

Comment:
lastseenrev is used to compare against w.rev to see if the event is already caught up with the store. see https://github.com/coreos/etcd/pull/5984/files#diff-32a7b4a028cc3e75ad34ce07813c1017R68